E2E fails on branching phase because expects branches that don't yet exist

As seen in https://github.com/submariner-io/releases/pull/189, when starting a stable release that requires a branching phase the E2E fails because it expect the soon-to-be-created branches to already exist.

```
[releases]$ _git checkout origin/release-0.10
[releases]$ git -C projects/submariner-operator checkout origin/release-0.10
error: pathspec 'origin/release-0.10' did not match any file(s) known to git
make: *** [Makefile:29: subctl] Error 1
```

--- a/Makefile
+++ b/Makefile
@@ -19,7 +19,10 @@ ifneq (,$(filter dryrun,$(_using)))
 override CREATE_RELEASE_ARGS += --dryrun
 endif
 
-TARGET_RELEASE = $(shell . scripts/lib/utils && determine_target_release 2&> /dev/null && echo $${file})
+E2E_NEEDED = $(shell . scripts/lib/utils && \
+    determine_target_release 2&> /dev/null && \
+    read_release_file && \
+    exit_on_branching && echo true)
 
 config-git:
 	git config --global user.email "$(GIT_EMAIL)";\
@@ -28,8 +31,12 @@ config-git:
 subctl: config-git
 	./scripts/subctl.sh $(SUBCTL_ARGS)
 
-e2e: $(if $(TARGET_RELEASE),deploy)
+ifeq (true, $(E2E_NEEDED))
+e2e: deploy
 	./scripts/e2e.sh
+else
+e2e: ;
+endif
 
 clusters: images subctl
 

--- a/scripts/images.sh
+++ b/scripts/images.sh
@@ -30,8 +30,5 @@ function pull_images() {
 
 determine_target_release
 read_release_file
-
-# If we're creating branches, no need to pull images as they won't exist and aren't needed yet anyhow
-[[ "${release['status']}" != "branch" ]] || exit 0
-
+exit_on_branching
 pull_images

--- a/scripts/lib/utils
+++ b/scripts/lib/utils
@@ -93,3 +93,7 @@ function extract_semver() {
     declare -gA semver
     IFS=".-" read -r semver['major'] semver['minor'] semver['patch'] semver['pre'] <<< "$1"
 }
+
+function exit_on_branching() {
+    [[ "${release['status']}" != "branch" ]] || exit 0
+}

